pypesto/profile/profile_next_guess.py

```diff
@@ -0,0 +1,280 @@
+        elif order == 1 or (np.isnan(order) and n_profile_points < 3):
+            # set the update direction
+            last_delta_x = current_profile.x_path[:, -1] - \
+                           current_profile.x_path[:, -2]
+            step_size_guess = np.abs(current_profile.x_path[par_index, -1] -
+                                     current_profile.x_path[par_index, -2])
+            delta_x_dir = last_delta_x / step_size_guess
+        elif np.isnan(order):
+            # disable rank warnings
+            # warnings.simplefilter('ignore', np.RankWarning)
+
+            # determine interpolation order
+            reg_max_order = np.floor(n_profile_points / 2)
+            reg_order = np.min([reg_max_order, options.reg_order])
+            reg_points = np.min([n_profile_points, options.reg_points])
+            reg_par = []
+            for i_par in range(0, len(problem.x_free_indices) + 1):
+                if i_par == pos_ind_red:
+                    reg_par.append(float('nan'))
+                else:
+                    # Do polynomial interpolation of profile path
+                    # Determine rank of polynomial interpolation
+                    regression_tmp = np.polyfit(
+                        current_profile.x_path[par_index, -1:-reg_points:-1],
+                        current_profile.x_path[i_par, -1:-reg_points:-1],
+                        reg_order, full=True)
+
+                    # Decrease rank if interpolation problem is illconditioned
+                    if regression_tmp[2] < reg_order:
+                        reg_order = regression_tmp[2]
+                        regression_tmp = np.polyfit(
+                            current_profile.x_path[par_index, -reg_points:-1],
+                            current_profile.x_path[i_par, -reg_points:-1],
+                            reg_order, full=True)
+
+                    reg_par.append(regression_tmp[0])
+
+    # boolean indicating whether a search should be carried out
+    search = True
+
+    # check whether we must make a minimum step anyway, since we're close to
+    # the next bound
+    min_delta_x = x[par_index] + par_direction * options.min_step_size
+    if par_direction is -1:
+        if min_delta_x < problem.lb_full[par_index]:
+            step_length = problem.lb_full[par_index] - x[par_index]
+            search = False
+    else:
+        if min_delta_x > problem.ub_full[par_index]:
+            step_length = problem.ub_full[par_index] - x[par_index]
+            search = False
+
+    if not search:
+        return x + step_length * delta_x_dir
+
+    # restrict a step to the bounds
+    def clip_to_bounds(step_proposal):
+        for i_par, i_step in enumerate(step_proposal):
+            step_proposal[i_par] = np.max([np.min([i_step, problem.ub_full[
+                i_par]]), problem.lb_full[i_par]])
+        return step_proposal
+
+    # parameter extrapolation function
+    def par_extrapol(step_length):
```

Comment:
Same for other nested functions. Do they need to be in here? 200 lines function body is somewhat long. Aim for < 10, excluding documentation.